be/src/storage/chunk_helper.cpp

```diff
@@ -373,4 +373,19 @@ std::shared_ptr<vectorized::Chunk> ChunkHelper::new_chunk(const std::vector<Slot
     return chunk;
 }
 
+void ChunkHelper::reorder_chunk(const TupleDescriptor& tuple_desc, vectorized::Chunk* chunk) {
+    return reorder_chunk(tuple_desc.slots(), chunk);
+}
+
+void ChunkHelper::reorder_chunk(const std::vector<SlotDescriptor*>& slots, vectorized::Chunk* chunk) {
+    DCHECK(chunk->columns().size() == slots.size());
+    auto reordered_chunk = vectorized::Chunk();
+    auto& original_chunk = (*chunk);
+    for (std::size_t idx = 0; idx < slots.size(); idx++) {
```

Comment:
IMO, not all slots will be read in chunk. So there will be slot in `slots` which has no corresponding column in chunk.
So I think you should continue rather than push a nullable in to reordered_chunk

Reply:
I have talked with @kangkaisen. He said if there is slot, there will be corresponding column